torch/distributed/_composable/replicate.py

```diff
@@ -22,6 +18,9 @@ def mark_modules(self, *modules: nn.Module) -> None:
             self.modules.append(module)
```

Comment:
add an assert len(self.modules)==1? as current codes can not support multiple replicate() calls, e.g., in the below codes, init_helper() can only be called once to collect the parameters for the first marked module.

Reply:
I think we don't need the assert?
Because `init_helper()` will check if it's been initialized to ensure it runs only once. So `_ReplicateState.mark_modules()` actually works for multiple modules.
`replicate()` simply calls `_ReplicateState.mark_modules()`. The reason it can be used on single module only is due to the current interface of `contract`, which accepts a single module.
Does this make sense to you?

Reply:
yeah, makes sense that contract() accepts a single module and you created a _ReplicateState() per module, so it will not have a case to have 2 modules, in this case why _ReplicateState.mark_modules() needs to support multiple modules?

Reply:
I'm keeping this because we can easily enable `replicate()` for multiple modules in the future, and that's also what we have in our original design doc. That will give us the benefit to group multiple small modules together to calculate buckets, and improve performance.

Reply:
sounds good, at that point, I think the per module registration needs change back as well?

Reply:
Yes, I think so